mne/export/_edf.py

```diff
@@ -53,7 +58,20 @@ def _export_raw(fname, raw, physical_range):
         stim_index = np.atleast_1d(stim_index.squeeze()).tolist()
         drop_chs.extend([raw.ch_names[idx] for idx in stim_index])
 
+    # add warning if any channel types are not voltage based
+    # note: we can write these other channels, such as 'misc'
+    # but they just won't work because EDF doesn't support
+    # the idea of a 'misc' channel
+    voltage_types = list(units) + ['stim', 'misc']
+    if any([ch not in voltage_types for ch in orig_ch_types]):
+        warn('There are non-voltage channels that are not set as "misc". '
+             'EDF only supports writing Voltage-based data. '
+             'These channels will not be written to the EDF file. '
+             'If you would still like these channels to be written, '
+             'set their channel type to "misc".')
```

Comment:
In addition to clarifying the warning, maybe we should list the channel names that cause this warning?

Reply:
Done